### ISSUE TITLE
Duplication in Exploit::Remote::Ftp and Rex::Proto::Ftp

### ISSUE THREAD

**Issue by pingport80** [rapid7/metasploit-framework#14915]
## Summary
Currently both `Exploit::Remote::Ftp` and `Rex::Proto::Ftp` are defining same methods, which could be reduced if former uses the methods defined in the later.

Useful Links :
      - [Rex::Proto::Ftp](https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/proto/ftp/client.rb)
      - [Exploit::Remote::Ftp](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/remote/ftp.rb)
 
`Exploit::Remote::HttpClient` mixin uses the core methods of HTTP protocol from `Rex::Proto::Http::Client`, but in case of FTP this is not happening which results in defining same methods in both the files.

**Comment by pingport80** [comment]
I would like to propose a library label for this issue.

**Comment by adfoster-r7** [comment]
Looks `Rex::Proto::Ftp` was was originally added with the intention of using it in the future: https://github.com/rapid7/metasploit-framework/pull/8762

**Comment by pingport80** [comment]
Yeah, I went through the discussion and support for the `rex::ftp` was discussed to be added in the other two implementations of ftp, but that didn't happen I guess. 

 `exploit::remote::ftp` can take methods from `rex::proto`, and also `metasploit/framework/ftp/client.rb` is used just in `Login_Scanner::Ftp` with almost the same methods. 
`Login_Scanner::Ftp` can take methods from `exploit::remote::ftp` or `rex` (as per requirement) without the need of `metasploit/framework/ftp/client.rb`.

It's not just the case of code duplication, as a new contributor when I was reading these libs, I started with taking FTP libs as an example because of it's simplicity, but actually it confused me and understood the concept later with HTTP libs.
